.github/workflows/docker-push.yaml

```diff
@@ -2,15 +2,13 @@ name: Docker Image CI
 
 on:
   push:
-    branches: [main]
+    branches: [preprod]
```

Comment:
Try to keep both branch, u will adapt your step that construct the tag according to the branch name